x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/field_parameters/inference_id_selects.tsx

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiCallOut, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React, { useEffect, useState } from 'react';
+
+import { useComponentTemplatesContext } from '../../../../component_templates/component_templates_context';
+import { getFieldConfig } from '../../../lib';
+import { FieldHook, Form, SuperSelectField, UseField, useForm } from '../../../shared_imports';
+import { SuperSelectOption } from '../../../types';
+interface Props {
+  onChange(value: string): void;
+  'data-test-subj'?: string;
+}
+
+export const InferenceIdSelects = ({ onChange, 'data-test-subj': dataTestSubj }: Props) => {
+  const { form } = useForm({ defaultValue: { main: 'elser_model_2' } });
+  const { subscribe } = form;
+  const { api } = useComponentTemplatesContext();
+  const [inferenceModels, setInferenceModels] = useState<any>([]);
```

Comment:
We shouldn't be using `any`. We know what this variable can be, so we should have a type for them.